Crash when trying to open deeplink

### Checklist

- [X] I have used the search function for [**OPEN**](https://github.com/AntennaPod/AntennaPod/issues) issues to see if someone else has already submitted the same bug report.
- [X] I have **also** used the search function for [**CLOSED**](https://github.com/AntennaPod/AntennaPod/issues?q=is%3Aissue+is%3Aclosed) issues to see if the problem is already solved and just waiting to be released.
- [X] I will describe the problem with as much detail as possible.
- [X] If the bug only to occurs with a certain podcast, I will include the URL of that podcast.

### App version

3.5.0

### Where did you get the app from

Google Play

### Android version

14

### Device model

_No response_

### First occurred

_No response_

### Steps to reproduce

1. Open the deeplink  https://antennapod.org/deeplink/subscribe/?url=%68%74%74ps%3A%2F%2Ffeeds.megaphone.fm%2FASD7453520464&title=Ralphs+Relevante+R%C3%A4tsel+%E2%80%93+Wissen+mit+Ralph+Caspers OR https://antennapod.org/deeplink/subscribe/ (I opened the link from Fedilab link card, which might strip parameters for privacy reasons)

### Expected behaviour

Open relevant podcast of if not possible, show a warning:
**Can't open podcast**
The link you tapped doesn't contain a valid podcast URL. Please verify the link and try again, or search for the podcast in AntennaPod.
CLOSE     OPEN SEARCH

### Current behaviour

App crash

### Logs

## Environment
Android version: 14
OS version: 5.4.242-qgki-g07bc4ccc612b
AntennaPod version: 3.5.0
Model: FP5
Device: FP5
Product: FP5

## Crash info
Time: 11-10-2024 07:53:05
AntennaPod version: 3.5.0

## StackTrace
```
java.lang.RuntimeException: Unable to start activity ComponentInfo{de.danoeh.antennapod/de.danoeh.antennapod.ui.screen.onlinefeedview.OnlineFeedViewActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.length()' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3783)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3923)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:139)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:96)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2444)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:205)
	at android.os.Looper.loop(Looper.java:294)
	at android.app.ActivityThread.main(ActivityThread.java:8223)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:977)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.length()' on a null object reference
	at java.net.URLDecoder.decode(URLDecoder.java:177)
	at java.net.URLDecoder.decode(URLDecoder.java:143)
	at de.danoeh.antennapod.net.common.UrlChecker.prepareUrl(SourceFile:61)
	at de.danoeh.antennapod.ui.screen.onlinefeedview.OnlineFeedViewActivity.onCreate(SourceFile:118)
	at android.app.Activity.performCreate(Activity.java:8595)
	at android.app.Activity.performCreate(Activity.java:8573)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1457)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3765)
	... 12 more
```

--- a/app/src/main/java/de/danoeh/antennapod/ui/screen/onlinefeedview/OnlineFeedViewActivity.java
+++ b/app/src/main/java/de/danoeh/antennapod/ui/screen/onlinefeedview/OnlineFeedViewActivity.java
@@ -102,15 +102,11 @@ public class OnlineFeedViewActivity extends AppCompatActivity {
             feedUrl = getIntent().getDataString();
         }
 
-        if (feedUrl == null) {
+        if (feedUrl == null || UrlChecker.isDeeplinkWithoutUrl(feedUrl)) {
             Log.e(TAG, "feedUrl is null.");
             showNoPodcastFoundError();
         } else {
             Log.d(TAG, "Activity was started with url " + feedUrl);
-            // Remove subscribeonandroid.com from feed URL in order to subscribe to the actual feed URL
-            if (feedUrl.contains("subscribeonandroid.com")) {
-                feedUrl = feedUrl.replaceFirst("((www.)?(subscribeonandroid.com/))", "");
-            }
             if (savedInstanceState != null) {
                 username = savedInstanceState.getString("username");
                 password = savedInstanceState.getString("password");

--- a/net/common/src/main/java/de/danoeh/antennapod/net/common/UrlChecker.java
+++ b/net/common/src/main/java/de/danoeh/antennapod/net/common/UrlChecker.java
@@ -62,12 +62,19 @@ public final class UrlChecker {
             } catch (UnsupportedEncodingException e) {
                 return prepareUrl(query);
             }
+        } else if (lowerCaseUrl.contains("subscribeonandroid.com")) {
+            return prepareUrl(url.replaceFirst("((www.)?(subscribeonandroid.com/))", ""));
         } else if (!(lowerCaseUrl.startsWith("http://") || lowerCaseUrl.startsWith("https://"))) {
             Log.d(TAG, "Adding http:// at the beginning of the URL");
             return "http://" + url;
         } else {
             return url;
         }
+    }
+
+    public static boolean isDeeplinkWithoutUrl(String url) {
+        return url.toLowerCase(Locale.ROOT).contains(AP_SUBSCRIBE_DEEPLINK)
+                && Uri.parse(url).getQueryParameter("url") == null;
     }
 
     /**

--- a/ui/i18n/src/main/res/values/strings.xml
+++ b/ui/i18n/src/main/res/values/strings.xml
@@ -317,7 +317,7 @@
     <string name="download_log_title_unknown">Unknown title</string>
     <string name="download_type_feed">Feed</string>
     <string name="download_type_media">Media file</string>
-    <string name="null_value_podcast_error">No podcast was provided that could be shown.</string>
+    <string name="null_value_podcast_error">The link you tapped does not contain a valid podcast URL. Please verify the link and try again, or search for the podcast manually.</string>
     <string name="no_feed_url_podcast_found_by_search">The suggested podcast did not have an RSS link, AntennaPod found a podcast that could match</string>
     <string name="authentication_notification_title">Authentication required</string>
     <string name="confirm_mobile_download_dialog_title">Confirm mobile download</string>

